lib/app.dart

```diff
@@ -28,51 +28,131 @@ library;
 
 // Group imports by dart, flutter, packages, local. Then alphabetically.
 
+import 'dart:io';
+
 import 'package:flutter/material.dart';
 
 import 'package:catppuccin_flutter/catppuccin_flutter.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 
 import 'package:rattle/home.dart';
 import 'package:rattle/r/start.dart';
+import 'package:rattle/constants/temp_dir.dart';
+import 'package:window_manager/window_manager.dart';
 
 // Add a key to reference [RattleHome] to access its method.
 
 final GlobalKey<RattleHomeState> rattleHomeKey = GlobalKey<RattleHomeState>();
 
+Future<void> cleanUpTempDirs() async {
+  // remove the temporary directories created by Rattle
+  // open from temp directory
+  final rattleTempDir = Directory(tempDir);
+  if (rattleTempDir.existsSync()) {
+    await rattleTempDir.delete(recursive: true);
+    debugPrint('Deleted Rattle temp directory: $tempDir');
+  }
+
+  // final rTempDir = Directory(rTempDir);
+  // for (var dir in rTempDir) {
+  //   if (dir is Directory) {
+  //     await dir.delete(recursive: true);
+  //     debugPrint('Deleted R temp directory: ${dir.path}');
+  //   }
+  // }
+}
+
 /// A widget for the root of the Rattle app encompassing the Rattle home widget.
 ///
-/// The root widget covers the screen of the app. This widget is stateless as it
-/// does not need to manage any state itself. The state is managed through
-/// riverpod and so it is a [ConsumerWidget].
-
-class RattleApp extends ConsumerWidget {
+/// This widget manages the application's lifecycle and handles cleanup
+/// operations when the app is about to close. It uses [ConsumerStatefulWidget]
+/// to interact with Riverpod providers and [WindowListener] to handle
+/// window-related events, particularly for desktop platforms.
+class RattleApp extends ConsumerStatefulWidget {
   const RattleApp({super.key});
 
-  /// Build the root widget as a [MaterialApp] widget, setting up the app theme,
-  /// and populating the widget with the Rattle home page widget.
+  @override
+  ConsumerState<RattleApp> createState() => _RattleAppState();
+}
 
+class _RattleAppState extends ConsumerState<RattleApp> with WindowListener {
+  /// Initializes the state and sets up window management.
```

Comment:
Empty line between comments and code.

Reply:
Out style is for a blank line after (and before most) comments.

Reply:
> Out style is for a blank line after (and before most) comments.

I will update that. BTW, are there any formatting tools or settings that could automatically handle them?

Reply:
Maybe some option in VSCode for this. Not sure though.

Reply:
The built-in dart format command, which is recommended in the Dart documentation, automatically removes the first line break between a class declaration and its first comment. Subsequent comments are not affected by this behavior.
Unlike clang-format, dart format cannot be customized. The Dart development team has stated this is intentional by design. For reference, see this GitHub issue: https://github.com/Dart-Code/Dart-Code/issues/914.
Given these limitations, I'm curious about which formatting tool you're using. Could you share that information? I'd like to adopt the same tool to ensure consistency in our codebase.
![image](https://github.com/user-attachments/assets/5ab88c76-95e2-4fff-acf3-d33d84707885)

Reply:
The image attached is ` dart format .`'s result

Reply:
I also noticed that if I ran `make prep`, `dart format .` runs.

Reply:
https://github.com/gjwgit/rattleng/blob/a1e295d468db0061d0b5d6b4f71fda01349b3a39/support/flutter.mk#L143

Reply:
> The built-in dart format command, which is recommended in the Dart documentation, automatically removes the first line break between a class declaration and its first comment. Subsequent comments are not affected by this behavior. 

Yes, `dart format` is opinionated! That's fine - We leave the first comment in a block without a blank line before it, but all others can have blanks around them. It makes the layout easier to read.